### DISCUSSION TITLE
FEAT: per-repo permissions with `/app/installations/{installation_id}/access_tokens` endpoint

### DISCUSSION THREAD

**Discussion by NickCrews** [community/community#36576] [Apps, API and Webhooks]
Hi! I have a feature request for [`POST /app/installations/{installation_id}/access_tokens`](https://docs.github.com/en/rest/apps/apps#create-an-installation-access-token-for-an-app).

Currently, you supply a list of repos and a list of requested permissions, and you get back a token with those permissions for all those repos.

I would like to be able to select permissions on a per-repo basis. For example for repo A I want permission X, and for repo B I want permission Y. Currently, I can only get X and Y for both A and B.

It doesn't work for us to do two separate API calls and get two separate tokens, because in our use case we can only supply one token to the downstream client. For example, I want to `pip install` directly from multiple github repos, but pip only supports using one access token at a time, so that single token needs to work for all the repos at the same time.

This is stemming from https://github.com/qoomon/github-actions-access-manager/issues/6#issuecomment-1281452817

I would imagine that in the resulting API, you would pass a list of (repo, permissions) pairs, instead of the single (repos, permissions) pair that you use now.

Is there somewhere better that I should post this? Or how do I know that the appropriate developers have seen this request?

**Comment by ossanna16** [comment]
Hey there @NickCrews and thank you for your suggestion! I'm glad you found our community 🙂
